frontend/app/src/components/helper/NftDetails.vue

```diff
@@ -51,9 +77,25 @@ const imageUrl = computed<string | null>(() => {
   return get(balanceData)?.imageUrl ?? '/assets/images/placeholder.svg';
 });
 
-const name = computed<string>(() => {
-  return get(balanceData)?.name ?? get(identifier);
+const name = computed<string | null>(() => {
+  const data = get(balanceData);
+  return data?.name || data?.collectionName || null;
 });
+
+const { isTaskRunning } = useTasks();
+const loading = isTaskRunning(TaskType.NF_BALANCES);
```

Comment:
We should also have a status update for the `NF_BALANCES` that uses `Section` can you use that instead? at least for the initial load?

Status covers all the action while task covers only the awaited part